### PR TITLE
Fix ChannelListController delegate funcs are not overridable

### LINKED ISSUE #1484: ChatChannelListVC doesn't properly filter channels
## What did you do?
Attempt to display ChatChannelListVC for only `.messaging` channels of which my user is a member

## What did you expect to happen?
Only `.messaging` channels that my user is a member of should be displayed

## What happened instead?
ChatChannelListVC displays some channels that are `.livestream` type

## GetStream Environment
**GetStream Chat version:** 4.0.1
**GetStream Chat frameworks:** StreamChat, StreamChatUI

## Additional context
See attached sample project:
[StreamChannelList.zip](https://github.com/GetStream/stream-chat-swift/files/7238763/StreamChannelList.zip)

Steps to reproduce:
- Open app
- Tap to 'Enter Channel List'. Notice, as expected, only 1 .messaging channel is displayed
- Tap Back and then tap 'Enter livestream1 channel'
- Wait for messages to render
- Tap Back and then tap 'Enter Channel List'
- Notice that now the .livestream channel is displayed in the channel list. This is the undesired behavior

### PR DESCRIPTION
**Pull request by b-onc**
....in ChannelListVC subclasses

Fixes #1484

### CODE REVIEW

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift [hunk 1]
```diff
@@ -279,6 +279,22 @@ open class ChatChannelListVC: _ViewController,
...
+        return dummy.controller(controller, shouldListUpdatedChannel: channel)
+    }
+    
+    open func controller(_ controller: ChatChannelListController, shouldListUpdatedChannel channel: ChatChannel) -> Bool {
+        // This is an ugly hack to call default protocol implementation
+        // see https://stackoverflow.com/a/32606899
+        class Dummy: ChatChannelListControllerDelegate {}
+        let dummy = Dummy()
+        return dummy.controller(controller, shouldListUpdatedChannel: channel)
+    }
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:296]: Can't we simply remove the default implementation? And add it to the `ChatChannelListVC` and also remove the `AnyChannelListControllerDelegate` ?

Comment (tbarbugli) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:296]: makes a lot sense to me

Comment (b-onc) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:296]: working on it, will update the PR shortly

**Review by nuno-vieira — APPROVED** [review]
LGTM

**Review by nuno-vieira — APPROVED** [review]

**Pull request merged**